pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/UpsertUtils.java

```diff
@@ -134,6 +137,38 @@ public void close()
     }
   }
 
+  public static class ComparisonColumnReader implements Closeable {
```

Comment:
Let's make 2 reader classes, one for single column and one for multiple columns to avoid impacting the performance for existing use cases